nanoc/lib/nanoc/filters/sass/importer.rb

```diff
@@ -36,13 +36,23 @@ def find(identifier, options)
       items = filter.items.find_all(identifier)
       return if items.empty?
 
-      content = if items.size == 1
-                  items.first.compiled_content
-                else
-                  items.map { |item| %(@import "#{item.identifier}";) }.join("\n")
-                end
+      syntax, content = if items.size == 1
+                          item = items.first
+
+                          [
+                            item.identifier.ext.downcase.to_sym || :scss,
+                            item.compiled_content,
+                          ]
+                        else
+                          import_all = items.map { |i| %(@import "#{i.identifier}";) }.join("\n")
+
+                          [
+                            :scss,
+                            import_all,
+                          ]
+                        end
```

Comment:
I started changing this file last week but didn't go as far as you did with updating tests.
I wrote something similar:
```
      content, syntax = if items.size == 1
                          [items.first.compiled_content, items.first.ext.downcase.to_sym || :sass]
                        else
                          [items.map { |item| %(@import "#{item.identifier}";) }.join("\n"), :scss]
                        end

      options[:syntax] = syntax
```
Which imho produces a smaller diff than yours.

I chose to make it `content, syntax` because that's somewhat what Sass does, see `find_relative()`.
I chose to make `:sass` the default syntax because that's also what Sass does.

Reply:
My primary goal is to write readable code, diff size doesn't matter much to me. Git stores snapshots anyways. 😉

The fallback is `:scss` because that's what's the original code was hard-coding. 

I needed to change the specs because they were failing now that Sass compiles the inferred syntax, but the content served by the cache was Scss. 

I'm open to exchanging `syntax` and `content` if that is preferred.

Reply:
Readable is a subjective matter. When I talk about minimizing the diff, I talk about the cognitive load of inspecting the diff, not Git's way of storing things. I let @ddfreyne decide.

Having `content, syntax` is symetric with what Sass does and what `#find_relative()` does.

As far as the current behavior of assuming `:scss`, it's a bug and I vaguely recall my first implementation before review inspected item's extension to decide the syntax. Maybe we could `#raise()` telling we don't know which syntax to pick?

Reply:
I don’t have much of an opinion here — if the tests pass and Rubocop is happy with it, then I’m happy.

Reply:
@gpakosz Made the changes as requested. I also added some new specs round importing partials.

Reply:
@agross Thanks for the update. Well my very own opinion is that I wouldn't have traded 5 lines of code in context for 3 additional private functions (see http://number-none.com/blow/john_carmack_on_inlined_code.html). Undersanting what's going on imho now requires chasing tiny functions doing little meaningful work.

So yeah I personally would have written it differently. But if both Rubocop and @ddfreyne are happy then I'm happy too!